npm/pkg/controlplane/translation/translatePolicy_test.go

```diff
@@ -2919,3 +2919,114 @@ func TestEgressPolicy(t *testing.T) {
 		})
 	}
 }
+
+func TestNpmLiteCidrPolicy(t *testing.T) {
+	// 3 tests
+	// Test 1) Npm lite enabled, CIDR + Namespace label Peers, returns error
+	// Test 2) NPM lite disabled, CIDR + Namespace label Peers, returns no error
+	// Test 3) Npm Lite enabled, CIDR Peers , returns no error
+
+	tests := []struct {
+		name           string
+		targetSelector *metav1.LabelSelector
+		peersFrom      []networkingv1.NetworkPolicyPeer
+		peersTo        []networkingv1.NetworkPolicyPeer
+		npmLiteEnabled bool
+		wantErr        bool
+	}{
+		{
+			name:           "peer nameSpaceSelector and ipblock in ingress rules",
```

Comment:
Thanks for writing tests. Could we add these cases as well?
- peer with both ipblock and selector
- no peers

Reply:
This is 3 peers:
```
[]networkingv1.NetworkPolicyPeer{
				{
					NamespaceSelector: &metav1.LabelSelector{
						MatchLabels: map[string]string{
							"peer-nsselector-kay": "peer-nsselector-value",
						},
					},
				},
				{
					IPBlock: &networkingv1.IPBlock{
						CIDR:   "172.17.0.0/16",
						Except: []string{"172.17.1.0/24", "172.17.2.0/24"},
					},
				},
				{
					IPBlock: &networkingv1.IPBlock{
						CIDR: "172.17.0.0/16",
					},
				},
			},
```

Each peer corresponds to an element of the YAML's list (more intuitively, a "section" denoted by a `-`) in a NetworkPolicy spec, where each element is OR'd together. Within a peer, the rules are AND'd. For more details, see "namespaceSelector and podSelector" under https://kubernetes.io/docs/concepts/services-networking/network-policies/#behavior-of-to-and-from-selectors

> single peer with both ipblock and selector

It sounds like this is actually impossible from the docs though. It is not one of the 4 kinds of to/from selectors. Do you mind verifying that it's impossible to create such a NetPol peer with ipblock and pod selector?

Reply:
Added test cases for:


- CIDR + label selector combo in same peer
- no peers 


for allow-all-internal case, added an exception check within the code for it